controllers/managedocs_controller.go

```diff
@@ -551,25 +572,73 @@ func (r *ManagedOCSReconciler) reconcileAlertmanager() error {
 	return nil
 }
 
-func (r *ManagedOCSReconciler) reconcileAlertmanagerConfig() error {
-	// r.Log.Info("Reconciling AlertmanagerConfig")
-	// _, err := ctrl.CreateOrUpdate(r.ctx, r.Client, r.alertmanagerConfig, func() error {
-	// 	if err := r.own(r.alertmanagerConfig); err != nil {
-	// 		return err
-	// 	}
+func (r *ManagedOCSReconciler) reconcileAlertmanagerConfigSecret() error {
+	r.Log.Info("Reconciling AlertmanagerConfig secret")
+
+	_, err := ctrl.CreateOrUpdate(r.ctx, r.Client, r.alertmanagerConfigSecret, func() error {
+		if err := r.own(r.alertmanagerConfigSecret); err != nil {
+			return err
+		}
+
+		var pagerdutyServiceKey string
 
-	// 	desired := utils.ObjectFromTemplate(templates.AlertmanagerConfigTemplate, r.Scheme).(*promv1a1.AlertmanagerConfig)
-	// 	r.alertmanagerConfig.ObjectMeta.Labels = map[string]string{monLabelKey: monLabelValue}
-	// 	// TODO: Setup pagerduty and deadman sqtich on the desired alertmanagr config
-	// 	r.alertmanagerConfig.Spec = desired.Spec
+		if err := r.get(r.pagerdutySecret); err != nil {
+			return fmt.Errorf("Unable to get pagerduty secret: %v", err)
+		}
 
-	// 	return nil
-	// })
-	// if err != nil {
-	// 	return err
-	// }
+		pagerdutySecretData := r.pagerdutySecret.Data
+		if len(pagerdutySecretData["PAGERDUTY_KEY"]) != 0 {
+			pagerdutyServiceKey = string(pagerdutySecretData["PAGERDUTY_KEY"])
+		}
+		if len(pagerdutySecretData["serviceKey"]) != 0 {
+			pagerdutyServiceKey = string(pagerdutySecretData["serviceKey"])
+		}
+		if pagerdutyServiceKey == "" {
+			return fmt.Errorf("Pagerduty service key is not defined")
+		}
 
-	return nil
+		alertmanagerConfig := struct {
+			Route struct {
+				GroupWait      string `yaml:"group_wait"`
+				GroupInterval  string `yaml:"group_interval"`
+				RepeatInterval string `yaml:"repeat_interval"`
+				Receiver       string `yaml:"receiver"`
+				Routes         [1]struct {
+					Match struct {
+						Severity string `yaml:"severity"`
+					} `yaml:"match"`
+					Receiver string `yaml:"receiver"`
+				} `yaml:"routes"`
+			} `yaml:"route"`
+			Receivers [1]struct {
+				Name             string `yaml:"name"`
+				PagerdutyConfigs [1]struct {
+					ServiceKey string `yaml:"service_key"`
+				} `yaml:"pagerduty_configs"`
+			} `yaml:"receivers"`
+		}{}
+
+		alertmanagerConfig.Route.GroupWait = "30s"
+		alertmanagerConfig.Route.GroupInterval = "5m"
+		alertmanagerConfig.Route.RepeatInterval = "4h"
+		alertmanagerConfig.Route.Receiver = "pagerduty"
+		alertmanagerConfig.Route.Routes[0].Match.Severity = "critical"
+		alertmanagerConfig.Route.Routes[0].Receiver = "pagerduty"
+		alertmanagerConfig.Receivers[0].Name = "pagerduty"
+		alertmanagerConfig.Receivers[0].PagerdutyConfigs[0].ServiceKey = pagerdutyServiceKey
```

Comment:
I would suggest refactoring this into a separate function, something like 
```
func GetAlertManagerConfig(pagerdutyServiceKey secret) runtime.Object {
```

As this code is very verbose and we are going to introduce more configuration into the config for alertmanager (for example DMS configuration)